apps/emqx_dashboard_sso/src/emqx_dashboard_sso_manager.erl

```
@@ -175,47 +184,60 @@ start_backend_services() ->
+        }),
+        Result
+    end,
+
+    ErrFun = fun({error, Reason} = Error) ->
+        SafeReason = emqx_utils:redact(Reason),
+        ?SLOG(error, #{
+            msg => "update_sso_failed",
+            backend => Backend,
+            reason => SafeReason
+        }),
+        Error
+    end,
+
+    %% we always make sure the valid configuration will update successfully,
+    %% ignore the runtime error during its update
     case emqx_conf:update(?MOD_KEY_PATH(Backend), UpdateReq, #{override_to => cluster}) of
         {ok, UpdateResult} ->
             #{post_config_update := #{?MODULE := Result}} = UpdateResult,
-            ?SLOG(info, #{
-                msg => "update_sso_successfully",
-                backend => Backend,
-                result => Result
-            }),
-            Result;
-        {error, Reason} ->
-            ?SLOG(error, #{
-                msg => "update_sso_failed",
-                backend => Backend,
-                reason => Reason
-            }),
-            {error,
-                case Reason of
-                    {_Stage, _Mod, Reason2} ->
-                        Reason2;
-                    _ ->
-                        Reason
-                end}
+            case Result of
+                ok ->
+                    OkFun(Result);
+                {ok, _} ->
+                    OkFun(Result);
+                {error, _} = Error ->
+                    ErrFun(Error)
+            end;
+        {error, _} = Error ->
+            ErrFun(Error)
     end.
 
 pre_config_update(_, {update, _Backend, Config}, _OldConf) ->
-    {ok, Config};
+    maybe_write_certs(Config);
 pre_config_update(_, {delete, _Backend}, undefined) ->
     throw(not_exists);
 pre_config_update(_, {delete, _Backend}, _OldConf) ->
     {ok, null}.
 
 post_config_update(_, UpdateReq, NewConf, _OldConf, _AppEnvs) ->
-    call({update_config, UpdateReq, NewConf}).
+    {ok, on_config_update(UpdateReq, NewConf)}.
```

if we are to ignore errors:
```suggestion
    _ = on_config_update(UpdateReq, NewConf),
    ok.
```